.buildkite/pipeline.yml

```diff
@@ -7,3 +7,11 @@ steps:
       NIX_REMOTE: daemon
     agents:
       system: x86_64-linux
+  - label: 'iohk-ops-shellcheck'
+    command: 'export NIX_PATH="nixpkgs=$(/run/current-system/sw/bin/nix-build "$(dirname -- $(readlink -f -- ${BASH_SOURCE[0]}))/fetch-nixpkgs.nix" -o nixpkgs)"; export NIX_BUILD_SHELL=/run/current-system/sw/bin/bash; /run/current-system/sw/bin/nix-shell -p shellcheck -p coreutils --run scripts/shellcheck.sh'
```

Comment:
Do you think it's worth factoring this?

There's actually a script in `cardano-sl` that implements something very close to what's needed: https://github.com/input-output-hk/cardano-sl/blob/develop/scripts/ci/nix-shell.sh